Assets/Scripts/Map/Generators/Patrolling/Partitioning/MeetingPoints/MeetingPointTimePartitionGenerator.cs

```diff
@@ -0,0 +1,137 @@
+// any later version.
+// 
+// MAES is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+// or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+// Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License along
+// with MAES. If not, see http://www.gnu.org/licenses/.
+// 
+// Contributors: 
+// Henrik van Peet,
+// Mads Beyer Mogensen,
+// Puvikaran Santhirasegaram
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Maes.Robot;
+
+using UnityEngine;
+
+namespace Maes.Map.Generators.Patrolling.Partitioning.MeetingPoints
+{
+    public class MeetingPointTimePartitionGenerator : IPartitionGenerator<HMPPartitionInfo>
+    {
+        private readonly PartitionGeneratorWithMeetingPoint _partitionGeneratorWithMeetings;
+        private PatrollingMap _patrollingMap = null!;
+        private TravelEstimator _travelEstimator = null!;
+
+        public MeetingPointTimePartitionGenerator(IPartitionGenerator<PartitionInfo> partitionGenerator)
+        {
+            _partitionGeneratorWithMeetings = new PartitionGeneratorWithMeetingPoint(partitionGenerator);
+        }
+
+        public void SetMaps(PatrollingMap patrollingMap, CoarseGrainedMap coarseMap, RobotConstraints robotConstraints)
+        {
+            _patrollingMap = patrollingMap;
+            _partitionGeneratorWithMeetings.SetMaps(patrollingMap, coarseMap, robotConstraints);
+            _travelEstimator = new TravelEstimator(coarseMap, robotConstraints);
+        }
+
+        public Dictionary<int, HMPPartitionInfo> GeneratePartitions(HashSet<int> robotIds)
+        {
+            var partitionsById = _partitionGeneratorWithMeetings.GeneratePartitions(robotIds);
+
+            var meetingPoints = FindSharedVertexMeetingPoints(partitionsById.Values.ToArray());
+
+            var hmpPartitionsById = new Dictionary<int, HMPPartitionInfo>();
+            foreach (var (robotId, partitionInfo) in partitionsById)
+            {
+                var robotMeetingPoints = meetingPoints
+                    .Where(m => m.RobotIds.Contains(robotId))
+                    .ToList();
+                hmpPartitionsById.Add(robotId, new HMPPartitionInfo(partitionInfo, robotMeetingPoints));
+            }
+
+            var globalMeetingIntervalTicks = hmpPartitionsById.Values.Select(EstimatePartitionMeetingIntervalTicks).Max();
+            var tickColorAssignment = new WelshPowellMeetingPointColorer(meetingPoints).Run();
+            foreach (var meetingPoint in meetingPoints)
+            {
+                meetingPoint.AtTicks = tickColorAssignment[meetingPoint.VertexId] * globalMeetingIntervalTicks;
+            }
+
+            return hmpPartitionsById;
+        }
+
+        private static MeetingPoint[] FindSharedVertexMeetingPoints(PartitionInfo[] partitions)
```

Comment:
Here you can return a tuple with the found meeting points information (without robotids) and then later in GeneratePartitions use this information to create the actual MeetingPoints with robotids included.